source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -19,6 +21,36 @@ namespace Extensions {
 namespace Clusters {
 namespace DynamicForwardProxy {
 
+namespace {
+constexpr absl::string_view DynamicHostFilterStateKey = "envoy.upstream.dynamic_host";
+constexpr absl::string_view DynamicPortFilterStateKey = "envoy.upstream.dynamic_port";
+
+class DynamicHostObjectFactory : public StreamInfo::FilterState::ObjectFactory {
+public:
+  std::string name() const override { return std::string(DynamicHostFilterStateKey); }
+  std::unique_ptr<StreamInfo::FilterState::Object>
+  createFromBytes(absl::string_view data) const override {
+    return std::make_unique<Router::StringAccessorImpl>(data);
+  }
+};
+class DynamicPortObjectFactory : public StreamInfo::FilterState::ObjectFactory {
+public:
+  std::string name() const override { return std::string(DynamicPortFilterStateKey); }
+  std::unique_ptr<StreamInfo::FilterState::Object>
+  createFromBytes(absl::string_view data) const override {
+    uint32_t port = 0;
+    if (absl::SimpleAtoi(data, &port)) {
```

Comment:
if the data supplied is invalid should we reject config instead of silently ignoring it?

Reply:
This is not config, it's used in Wasm ABI at runtime. It's not silent, it should be translated to BAD_ARGUMENT status code if the output is nullptr.